### PR TITLE
buildbot: 2.8.4-> 2.9.2

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Updates buildbot to the latest version: https://github.com/buildbot/buildbot/releases/tag/v2.9.0

pyjade has been replaced with pypugjs. This is mostly just a rename due to trademark issues, but pypugjs is also better maintained than pyjade was. I removed pyjade because buildbot was the only user and it is obsolete now.

cc @nand0p @ryansydnor

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by veprbl — APPROVED** [review]
LGTM

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5181,8 +5181,6 @@ in {
 
   pyipp = callPackage ../development/python-modules/pyipp { };
 
-  pyjade = callPackage ../development/python-modules/pyjade { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:5184]: Do we need to add an alias which throws the removal?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
  </ul>
</details>
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>buildbot-worker (python38Packages.buildbot-worker)</li>
    <li>python37Packages.buildbot</li>
    <li>python37Packages.buildbot-full</li>
    <li>python37Packages.buildbot-pkg</li>
    <li>python37Packages.buildbot-plugins.console-view</li>
    <li>python37Packages.buildbot-plugins.grid-view</li>
    <li>python37Packages.buildbot-plugins.waterfall-view</li>
    <li>python37Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python37Packages.buildbot-plugins.www</li>
    <li>python37Packages.buildbot-ui</li>
    <li>python37Packages.buildbot-worker</li>
    <li>python37Packages.pypugjs</li>
    <li>python38Packages.buildbot-pkg</li>
    <li>python38Packages.buildbot-plugins.console-view</li>
    <li>python38Packages.buildbot-plugins.grid-view</li>
    <li>python38Packages.buildbot-plugins.waterfall-view</li>
    <li>python38Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python38Packages.buildbot-plugins.www</li>
    <li>python38Packages.pypugjs</li>
    <li>python39Packages.buildbot-pkg</li>
    <li>python39Packages.buildbot-plugins.console-view</li>
    <li>python39Packages.buildbot-plugins.grid-view</li>
    <li>python39Packages.buildbot-plugins.waterfall-view</li>
    <li>python39Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python39Packages.buildbot-plugins.www</li>
    <li>python39Packages.buildbot-worker</li>
    <li>python39Packages.pypugjs</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105935` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>buildbot-worker (python38Packages.buildbot-worker)</li>
    <li>python37Packages.buildbot</li>
    <li>python37Packages.buildbot-full</li>
    <li>python37Packages.buildbot-ui</li>
    <li>python37Packages.buildbot-worker</li>
    <li>python39Packages.buildbot</li>
    <li>python39Packages.buildbot-full</li>
    <li>python39Packages.buildbot-ui</li>
    <li>python39Packages.buildbot-worker</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>python37Packages.buildbot-pkg</li>
    <li>python37Packages.buildbot-plugins.console-view</li>
    <li>python37Packages.buildbot-plugins.grid-view</li>
    <li>python37Packages.buildbot-plugins.waterfall-view</li>
    <li>python37Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python37Packages.buildbot-plugins.www</li>
    <li>python37Packages.pypugjs</li>
    <li>python38Packages.buildbot-pkg</li>
    <li>python38Packages.buildbot-plugins.console-view</li>
    <li>python38Packages.buildbot-plugins.grid-view</li>
    <li>python38Packages.buildbot-plugins.waterfall-view</li>
    <li>python38Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python38Packages.buildbot-plugins.www</li>
    <li>python38Packages.pypugjs</li>
    <li>python39Packages.buildbot-pkg</li>
    <li>python39Packages.buildbot-plugins.console-view</li>
    <li>python39Packages.buildbot-plugins.grid-view</li>
    <li>python39Packages.buildbot-plugins.waterfall-view</li>
    <li>python39Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python39Packages.buildbot-plugins.www</li>
    <li>python39Packages.pypugjs</li>
  </ul>
</details>

**Comment by lopsided98** [comment]
2.9.1 has been released, so I have updated this PR.

**Comment by FRidh** [comment]
@GrahamcOfBorg build buildbot

**Comment by lopsided98** [comment]
I have updated this PR to include pypugjs 5.9.7 and buildbot 2.9.2.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>buildbot (python38Packages.buildbot)</li>
    <li>buildbot-full (python38Packages.buildbot-full)</li>
    <li>buildbot-ui (python38Packages.buildbot-ui)</li>
    <li>buildbot-worker (python38Packages.buildbot-worker)</li>
    <li>python37Packages.buildbot</li>
    <li>python37Packages.buildbot-full</li>
    <li>python37Packages.buildbot-pkg</li>
    <li>python37Packages.buildbot-plugins.console-view</li>
    <li>python37Packages.buildbot-plugins.grid-view</li>
    <li>python37Packages.buildbot-plugins.waterfall-view</li>
    <li>python37Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python37Packages.buildbot-plugins.www</li>
    <li>python37Packages.buildbot-ui</li>
    <li>python37Packages.buildbot-worker</li>
    <li>python37Packages.pypugjs</li>
    <li>python38Packages.buildbot-pkg</li>
    <li>python38Packages.buildbot-plugins.console-view</li>
    <li>python38Packages.buildbot-plugins.grid-view</li>
    <li>python38Packages.buildbot-plugins.waterfall-view</li>
    <li>python38Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python38Packages.buildbot-plugins.www</li>
    <li>python38Packages.pypugjs</li>
    <li>python39Packages.buildbot-pkg</li>
    <li>python39Packages.buildbot-plugins.console-view</li>
    <li>python39Packages.buildbot-plugins.grid-view</li>
    <li>python39Packages.buildbot-plugins.waterfall-view</li>
    <li>python39Packages.buildbot-plugins.wsgi-dashboards</li>
    <li>python39Packages.buildbot-plugins.www</li>
    <li>python39Packages.buildbot-worker</li>
    <li>python39Packages.pypugjs</li>
  </ul>
</details>

**Comment by lopsided98** [comment]
One more update, pypugjs 5.9.8 allows us to fetch from PyPI. All of the issues I ran into with pypugjs have now been resolved upstream and released.

**Pull request merged**